Sources/SpeziViews/Environment/Logger.swift

```diff
@@ -0,0 +1,31 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import OSLog
+import SwiftUI
+
+/// An `EnvironmentKey` to be used for providing a specific `Logger` object to subviews.
+///
+/// This might be useful to provide a logger to generic subviews depending on the context they are being used in.
+private struct LoggerKey: EnvironmentKey {
+    static let defaultValue: Logger = .init()
```

Comment:
I am wondering if we should provide a default subsystem and category for the logger to be easily identifiable across logging output. E.g. a `edu.stanford.spezi` subsystem and `views` category?

Reply:
I was thinking about this as well, but I decided against it, since the property is simply called `logger`, so the association with Spezi might not be obvious which makes the default app-specific Logger object more intuitive imho. I don't have a strong opinion though, so if you do, I will change it.